---
title: "Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R"
date: "`r format(Sys.time(), '%d-%m-%Y, %H:%M')`"
output: pdf_document
params:
  data: patient
---


```{r setup, include=FALSE}
#knitr::opts_knit$set(root.dir = "/media/vera/big_data/coriandR/output/")
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
#knitr::opts_chunk$set(dev = "png", dev.args = list(type = "cairo-png"))
knitr::opts_chunk$set(dev = 'pdf')

library(knitr)
```


```{r patient meta data, echo = FALSE}
patient.meta = data.frame(t(read.table(paste0(as.character(params$data), ".meta.tsv"), stringsAsFactors = F, row.names=1)))
if(tolower(patient.meta$gender) %in% c("f", "female")) {
  patient.meta$gender = "f"
} else {
  patient.meta$gender = "m"
}
```


## Whole genome sequencing depth of patient data

```{r data definition, echo = FALSE, fig.width=7, fig.height=3}
# Read patient counts from featureCounts output
patient = read.table(as.character(patient.meta$count_data), header=TRUE, row.names = 1, stringsAsFactors = F)
patient = patient[patient$Chr != 'chrM', ]

# Read PON table
pon = read.table(as.character(patient.meta$pon), header=TRUE, row.names = 1, stringsAsFactors = F)

# Make PON gender specific for patient and concatenate the data
if(patient.meta$gender == "f") {
  ponXf = pon[pon$Chr == "chrX_F", ]
  ponXf$Chr = "chrX"
  pon = rbind(pon[pon$gender == "neutral", ], ponXf)
  pon$patient = patient[patient$Chr != "chrY", 6]
} else {
  ponXm = pon[pon$Chr == "chrX_M", ]
  ponXm$Chr = "chrX"
  ponYm = pon[pon$Chr == "chrY_M", ]
  ponYm$Chr = "chrY"
  pon = rbind(pon[pon$gender == "neutral", ], ponXm, ponYm)
  pon$patient = patient[, 6]
}
pon$bin = 1:nrow(pon)  # needs recounting after gendering

# Calculate normalized patient counts relative to median sequencing depth and multiply by expected ploidy
pon$patient.norm = pon$patient / median(pon$patient)


# Giemsa-bands for plotting
# The bands table contains the g-bands of the human genome. Cheung VG, Nowak N, Jang W, Kirsch IR, Zhao S, Chen XN, Furey TS, Kim UJ, Kuo WL, Olivier M et al. Integration of cytogenetic landmarks into the draft sequence of the human genome. Nature. 2001 Feb 15;409(6822):953-8. PMID: 11237021
cytobands = read.table("../../tables/cytobands_sort.2.tsv", header=TRUE, sep = "\t", row.names = 1, stringsAsFactors = F)
band_types = data.frame(hue=c(0.3, 1, 1, 1, 1, 1, 0.1, 0.8), saturation=c(1, 0, 0, 0, 0, 0, 1, 1), value=c(1, 1, 0.5, 0.8, 0.7, 0.6, 1, 1), row.names=c("acen", "gneg", "gpos100", "gpos25", "gpos50", "gpos75", "gvar", "stalk"))
cytobands$hue = band_types[cytobands$Band.type, "hue"]
cytobands$saturation = band_types[cytobands$Band.type, "saturation"]
cytobands$value = band_types[cytobands$Band.type, "value"]


# List of relevant cancer genes
cancer_genes = read.table("../../tables/cancer.driver.genes_unique.and.goi.aml.csv", header=TRUE, sep = "\t", stringsAsFactors = F)

# List of all genes in human genome
all_genes = read.table("../../tables/all_genes.tsv", header=TRUE, row.names = 1, stringsAsFactors = F)
all_genes$start = as.numeric(all_genes$start)
all_genes$end = as.numeric(all_genes$end)

# List of chromosomes for plotting
contigs = data.frame(pos=tapply(pon$bin, pon$Chr, mean))
contigs$chr = gsub("chr", "", rownames(contigs))
contigs = contigs[order(as.numeric(contigs$chr)),]
par(mar=c(2, 4, 1, 1))
plot(pon$bin, pon$patient, ylim = c(0, qnorm(0.999, mean(pon$patient), sd(pon$patient))),
     col = "#11111111",
     pch = 16, xaxt="n", xlab="", las = 1, ylab="Read depth", xaxs="i")
abline(v = c(pon$bin[pon$Start == 1], nrow(pon)), col="#55555555", cex = 0.5)
with(contigs[(1:(nrow(contigs)/2))*2, ], axis(side=1, labels=chr, at=pos, line=-.5, cex.axis=0.5, tick=F))
with(contigs[(1:(nrow(contigs)/2))*2-1, ], axis(side=1, labels=chr, at=pos, line=-1, cex.axis=0.5, tick=F))
```


## Sample sequencing characteristics and mapping statistics

```{r mapping statistics, echo = FALSE}
stats = data.frame(t(read.table(as.character(patient.meta$mapping_stats), stringsAsFactors = F, row.names=1)))
stats$unique_mapping_pairs = round(sum(patient[ , 6]/2))

stats$raw_read_pairs = format(stats$raw_read_pairs, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
stats$average_read_length = format(stats$average_read_length, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
stats$unique_mapping_pairs = format(stats$unique_mapping_pairs, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)

```

| Sample characterisic | Value |
|---------------------:|:------|
| **Sample name**      | **`r as.character(patient.meta$name)`**     |
| **Sample gender**    | **`r as.character(patient.meta$gender)`**   |
| Raw read pairs       | `r as.character(stats$raw_read_pairs)`      |
| Average read length  | `r as.character(stats$average_read_length)` |
| Unique mapping pairs | `r as.character(stats$unique_mapping_pairs)`|


## Distribution of sequencing depth per bin in sample
```{r echo = FALSE, fig.width=7, fig.height=4}
with(pon[!pon$blocked, ], 
     {
      hist(patient, breaks=seq(0, max(patient)+20, by=20),
           xlim=c(0, median(patient)*2),
           freq=F, xlab="Depth per bin", border=T, col="blue", main="")
      pon.hist = hist(norm.clean.median*median(patient), breaks=seq(0, max(norm.clean.median*median(patient)*2,  na.rm=T), by=20), plot=F)
      lines(pon.hist$mids, pon.hist$density, col="red")
      abline(v = c(0.5, 1, 1.5, 2) *median(patient), col='gray', lty=2)
      legend("topright", legend="scaled PON", col="red", lty=1, cex=0.8)
      mtext(c("x1", "x2", "x3", "x4"), at=c(0.5, 1, 1.5, 2) *median(patient), side=3)
    })
```

This graph shows the distribution of the bins according to the depth of the sequencing. The red curve represents the distribution of the PON scaled to the same depth. Ideally, the histogram and the red curve match completely. If not, this can indicate larger deletions (the first peak decreases) or larger amplifications (the third peak increases). The indications "x1, x2, x3, x4" indicate the number of copies. 

\newpage

```{r gauss-test, echo = FALSE, fig.width=8, fig.height=3}
sd = mean(unlist(pon$norm.clean.sd), na.rm=T)
pon$pat.z = (pon$patient.norm-1) / sd
pat.dens = density(pon$pat.z, na.rm=T)

# Presentation of patient Z-values compared to estimated PON distribution
plot(seq(from=-20, to=20, by=0.1), dnorm(seq(from=-20, to=20, by=0.1)), type='l', col="gray",
     main="Patient Z-values compared to estimated PON distribution", ylab="Density", xlab="Z-value", ylim=c(0, max(pat.dens$y)))
#lines(density((pon$norm.clean.mean-1)/sd, na.rm=T), col="blue")
lines(pat.dens, col="black")
legend("topright", legend=c("theoretical Z-Distribution", "Patient"), col=c("gray", "black"), lty=1, cex=0.7)


# Detection of all bins that are outside the normal distribution after applying a Gauss test and after p-value adjustment according to the Benjamini-Hochberg method 
pon$gauss = apply(pon[, c("patient.norm", "norm.clean.median", "norm.clean.sd")], 1, function(x) ifelse(x[1] == 0, 1, min(pnorm(q=x[1], mean=x[2], sd=x[3], lower.tail=x[1]<x[2])*2, 1)))
pon$gauss.adjust = p.adjust(pon$gauss, method="hochberg")
pon$gauss.adjust[is.na(pon$gauss.adjust)] = 1
par(mar=c(2, 4, 1, 1))
with(pon[!pon$blocked, ],
     plot(bin, ifelse(patient.norm > 2, 4, patient.norm*2), ylim=c(0, 4),
          col = ifelse(gauss.adjust < 0.05, "#ff000033", "#11111111"),
          pch = ifelse(patient.norm > 2, 17, 20), xaxt="n", xlab="",
          main="Overview of the bins deviating from the normal distribution", las = 1, ylab="Ploidy", xaxs="i")
)

abline(v = c(pon$bin[pon$Start == 1], nrow(pon)), col="#55555555", cex = 0.5)
with(contigs[(1:(nrow(contigs)/2))*2, ], axis(side=1, labels=chr, at=pos, line=-.5, cex.axis=0.5, tick=F))
with(contigs[(1:(nrow(contigs)/2))*2-1, ], axis(side=1, labels=chr, at=pos, line=-1, cex.axis=0.5, tick=F))
```

In this graphic, the bins are shown in red that are outside the normal distribution after applying a Gauss test and after p-value adjustment according to the Benjamini-Hochberg method. 


```{r abberation characters, echo = FALSE}
pon$aberration = factor(ifelse((pon$patient.norm > 1.5 *pon$norm.clean.median & pon$gauss.adjust < 0.05), 'amp', ifelse((pon$patient.norm < 0.75 *pon$norm.clean.median & pon$gauss.adjust < 0.05), 'del', 'normal')), levels = c('normal', 'amp', 'del'))
```

```{r gc content correlation, echo = FALSE}
plot(pon$patient.norm/pon$norm.clean.median, pon$gc, type='p',
     main="Correlation ratio patient/median pon to gc content", pch = 16,
     xlab="Patient/median(PON)", ylab="GC-Content", col="#11111111")
```


\newpage

## Calculated numerical karyotype:
```{r transform of bins to giemsa bands and create of digital karyotype, echo = FALSE}
cytobands$aberration = factor(apply(cytobands, 1, function(x) {
  bins = pon[pon$Chr == x[1] & pon$Start >= floor(as.integer(x[2])/10^6)*10^6 & pon$End <= ceiling(as.integer(x[3])/10^6)*10^6 & !pon$blocked,]
  if(nrow(bins) < 1) return("normal")
  if(sum(bins$aberration == "del")/nrow(bins) > 0.5) return("del")
  if(sum(bins$aberration == "amp")/nrow(bins) > 0.5) return("amp")
  return("normal")
}))

cytobands$hue[cytobands$aberration == "amp"] = 0.65
cytobands$saturation[cytobands$aberration == "amp"] = 1
cytobands$hue[cytobands$aberration == "del"] = 0
cytobands$saturation[cytobands$aberration == "del"] = 1

# bands merging
abnorm.bands = cytobands[cytobands$aberration != "normal", ]
colnames(abnorm.bands)[colnames(abnorm.bands) == "Name"] = "Start.band"
karyotype_to_report = abnorm.bands[1, ]

if (nrow(abnorm.bands) >= 2) {
  k = 1
  # the counter in table karyotype_to_report
  for (s in 2:nrow(abnorm.bands)) {
    if (karyotype_to_report$Chr[k] == abnorm.bands$Chr[s] &
        karyotype_to_report$aberration[k] == abnorm.bands$aberration[s] &
        karyotype_to_report$End[k] == abnorm.bands$Start[s]) 
      { 
        karyotype_to_report$End.band[k] = abnorm.bands$Start.band[s]
        karyotype_to_report$End[k] = abnorm.bands$End[s]
        k = nrow(karyotype_to_report)
    } else {
       karyotype_to_report[k+1, ] = abnorm.bands[s, ]
       karyotype_to_report[k+1, "End.band"] = karyotype_to_report[k+1, "Start.band"]
       # add the band on a new chr or with a different aberration
       k = nrow(karyotype_to_report)
       # this is now the counter band
    }
  }
  for (g in 1:nrow(karyotype_to_report)) {
    if (is.na(karyotype_to_report$End.band[g])) {
      karyotype_to_report$End.band[g] <- as.character(karyotype_to_report$Start.band[g])
    }
  }
} 

karyotype_to_report$karyotype = as.character(paste0(karyotype_to_report$aberration, "(",
                                                       sub("chr", "", karyotype_to_report$Chr), ")(",
                                                       karyotype_to_report$Start.band,
                                                       karyotype_to_report$End.band, ")"))

kable(karyotype_to_report$karyotype, col.names = NULL, align = "l", "html")
```


\newpage
## Genes affected by CNVs\*:
```{r genes in aberration bins table, echo = FALSE}
  special_genes = all_genes[all_genes$symbol %in% cancer_genes$genes, ]
  special_genes$aberration = 'norm'
  special_genes$found = 0
  
  for(i in 1:nrow(special_genes)) {
    genes.in.bin = pon[(pon$aberration != "normal" & pon$Chr == special_genes$chr[i] & pon$Start <= special_genes$start[i] & special_genes$start[i] <= pon$End) |
                       (pon$aberration != "normal" & pon$Chr == special_genes$chr[i] & pon$Start <= special_genes$end[i] & special_genes$end[i] <= pon$End), ]
    
    if(nrow(genes.in.bin) == 0) {next}
    special_genes$found[i] = 1
    if(all(genes.in.bin$aberration == 'amp')) {special_genes$aberration[i] = 'amp'}
    if(all(genes.in.bin$aberration == 'del')) {special_genes$aberration[i] = 'del'}
  }
  
reported_genes = special_genes[special_genes$found == 1 & (special_genes$aberration == "amp" | special_genes$aberration == "del"), ]
reported_genes$start = format(reported_genes$start, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
reported_genes$end = format(reported_genes$end, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
  
kable(reported_genes[ , 1:6], align = "lllrrl")
```

> Bailey et al. (2018): Comprehensive Characterization of Cancer Driver Genes and Mutations. Cell 173 (2), 371-385.e18. DOI: 10.1016/j.cell.2018.02.060.;

> Papaemmanuil et al. (2016): Genomic Classification and Prognosis in Acute Myeloid Leukemia. The New England journal of medicine 374 (23), S. 2209â€“2221. DOI:10.1056/NEJMoa1516192.

\newpage

## Chromosome overview plots:

```{r simplePlots, echo = FALSE, fig.width=12, fig.height=4}
#units = c('0', '00', 'K', '0K', '00K', 'M', '0M', '00M', 'B', 'B', 'B', 'T', 'T', 'T')
par(mar=c(4, 4, 0.5, 5.5))

for(i in unique(pon$Chr)) {
  with(pon[pon$Chr == i,], {
    plot(Start, ifelse(patient.norm > 2.5, 4, patient.norm*2), col = ifelse(blocked, 0, c("black", "blue", "red")[aberration]), ylim=c(0,5), xlim=c(0,max(End)), xaxt='n', xaxs="i", xlab='', ylab='Ploidy', pch=ifelse(patient.norm > 2.5, 17, 20), las=2)
    segments(Start, (norm.clean.median+norm.clean.sd*3)*2, Start, (norm.clean.median-norm.clean.sd*3)*2, col="#aaaaaaaa")
    #mag = floor(log10(max(End)))
    #xmax = floor(max(End)/10^mag+1)*10^mag
    axis(1, seq(0, 10^9, by=2*10^7), paste(seq(0, 1000, by=20), "Mbp"), line=2, tck=-0.05)
    axis(1, seq(0, 10^9, by=10^7), rep("", 10^9/10^7+1), line=2)
    #axis(1, seq(0, mag+2)*10^(mag), c('0 bp', paste0(seq(1, mag+2), units[mag], ' bp')), line=2)
  })
  mtext(i, side=2, line=0.9, at=-0.4, las=2, cex=1.5)
  abline(h=0:4, lty=3, col="#aaaaaaaa")
  with(cytobands[cytobands$Chr == i, ], {
    rect(Start, -.25, End, -.55, xpd=T,
         col=hsv(hue, saturation, value, 0.3)
         )
  })
  legend("right", c("amplification", "gain*", "deletion", "stalk", "centromer"), fill=c("white", "white", "white", "#aaaa0055", "#00990055"), col=c("blue", "blue", "red", NA, NA), xpd=T, inset=c(-.1, 0), border=F, lty=0, pch=c(20, 2, 20, NA, NA), cex=0.7)
}
```

\newpage

\* The reads that are marked with triangles have a particularly deep sequencing depth, i.e. there are more than 4 copies.

\*\* The legend contains characteristics to Giemsa stain results. Recognized stain values: gneg, gpos50, gpos75, gpos25, gpos100, acen, gvar, stalk.

> Cheung VG, Nowak N, Jang W, Kirsch IR, Zhao S, Chen XN, Furey TS, Kim UJ, Kuo WL, Olivier M et al. Integration of cytogenetic landmarks into the draft sequence of the human genome. Nature. 2001 Feb 15;409(6822):953-8. PMID: 11237021



