---
title: "Report ´coriandR: ChrOmosomal abeRration Identifier AND Reporter in R´"
subtitle: "Report sample"
date: "`r format(Sys.time(), '%d-%m-%Y')`"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_chunk$set(dev = "png", dev.args = list(type = "cairo-png"))
```

```{r sample data, echo = FALSE}
patient.data = read.table("patient.data.tsv", stringsAsFactors = F)
patient_name = patient.data[1, 2]
gender = as.character(patient.data[2, 2])
```

```{r data definition, echo = FALSE}
library(ggplot2)
library(knitr)

patient = read.table("patient.fc.tsv", header=TRUE, row.names = 1, stringsAsFactors = F) 
bin_data = read.table("pon.tsv", header=TRUE, row.names = 1, stringsAsFactors = F)
bin_data$Start = as.integer(bin_data$Start)
bin_data$End = as.integer(bin_data$End)

# gc-content, created with bedtools
gc = read.table("GRCh38.p13.genome.1M.nucl", stringsAsFactors = T)
pon$gc = gc[gc$V1 != 'chrM', 5]

#The bands table contains the g-bands of the human genome. Cheung VG, Nowak N, Jang W, Kirsch IR, Zhao S, Chen XN, Furey TS, Kim UJ, Kuo WL, Olivier M et al. Integration of cytogenetic landmarks into the draft sequence of the human genome. Nature. 2001 Feb 15;409(6822):953-8. PMID: 11237021
cytobands = read.table("../../tables/cytobands_sort.2.tsv", header=TRUE, sep = "\t", row.names = 1, stringsAsFactors = F)

cancer_genes = read.table("../../tables/cancer.driver.genes_unique.and.goi.aml.csv", header=TRUE, sep = "\t", stringsAsFactors = F)
all_genes = read.table("../../tables/all_genes.tsv", header=TRUE, row.names = 1, stringsAsFactors = F)
all_genes$start = as.integer(all_genes$start)
all_genes$end = as.integer(all_genes$end)
```

```{r mapping stats, echo = FALSE}
stats = read.table("mapping.stats.tsv", stringsAsFactors = F)
stats[nrow(stats) +1, ] = list(as.character("unique_mapped_pairs"), round(sum(patient[ , 6])/2))
stats[nrow(stats) +1, ] = list(as.character("gender"), gender)

stats[ , 1] = c("row read pairs", "average read length", "unique mapped pairs", "gender")

library(knitr)
kable(stats, digits = 0, col.names = NULL, align = "r")
```


## Distribution of bin depth in sample
The following plot shows the distribution of the bins according to the depth of the sequencing. The blue curve represents the distribution of the bins in the PON.

Optimally the histogram and the blue curve match completely. If not, this can indicate larger deletions (the first peak decreases) or larger amplifications (the third peak increases). 

```{r Distribution of bin depth in sample, echo = FALSE}
ggplot(data = patient, mapping = aes(x = patient[ , 6])) +
         geom_histogram(aes(y = stat(density)), binwidth = 10) +
         geom_density(data = bin_data, aes(x = mean * median(patient[, 6], na.rm = TRUE)/2), color = "steelblue3") +
         xlim(0, median(patient[, 6])*2.5) +
         xlab(" ") +
         theme_classic()
```

```{r pon gender adjust, echo = FALSE}
# depending on the gender of the sample, only the same sex chromosomes are included in the calculation
pon.gender.adjust = function(pon, gender) {
  if (gender == "M") {
    chrn = "chrX"
    pon.copy = pon[pon$Chr == "chrX_M", ]
    pon.copy$Chr = chrn
    pon = rbind(pon, pon.copy)
    
    chrn = "chrY"
    pon.copy = pon[pon$Chr == "chrY_M", ]
    pon.copy$Chr = chrn
    pon = rbind(pon, pon.copy)
    
    pon = pon[!(pon$Chr == "chrX_F" | pon$Chr == "chrX_M" | pon$Chr == "chrY_M"), ]
  }
  
  else {
    chrn = "chrX"
    pon.copy = pon[pon$Chr == "chrX_F", ]
    pon.copy$Chr = chrn
    pon = rbind(pon, pon.copy)

    pon = pon[!(pon$Chr == "chrX_F" | pon$Chr == "chrX_M" | pon$Chr == "chrY_M"), ]
  }
}
```

```{r Normalizing sample data by library size and per megabase, echo = FALSE}
#Normalizing to a ploidity of 2, the PON looks much more straight forward aside from the samples with a second mode at 1 (gonosomes).
pon.norm = pon[, meta$pon.col]
#pon.norm = data.frame(t(t(pon.norm)/ apply(pon.norm, 2, median, na.rm=T)*2))
pon.norm = data.frame(t(t(pon.norm)/(apply(pon.norm, 2, sum, na.rm=T))))
pon.norm = data.frame(pon.norm/apply(pon.norm, 1, median, na.rm=T)*2)

patient_normalization = function(patient, column_number) {
  patient_norm <- patient[ , 1:5]
  patient_norm$pat_bins <- data.frame(t(t(patient[ , column_number])/apply(patient[ , column_number], 1, median, na.rm=T)*2))
  return(patient_norm)
}
```


\newpage

```{r test, echo = FALSE}
bin_data = read.table("/media/vera/big_data/mr_genoms/pon.marburg.compliete/pon.tsv", header=TRUE, row.names = 1, stringsAsFactors = F)
```




