\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{18 MÃ¤rz, 2021}


\begin{document}
\maketitle

\hypertarget{sample-id}{%
\subsection{Sample ID}\label{sample-id}}

\begin{verbatim}
## 109516.new.report
\end{verbatim}

\begin{longtable}[]{@{}rr@{}}
\toprule
\endhead
row read pairs & 3767290\tabularnewline
average read length & 145.304\tabularnewline
unique mapped pairs & 3617856\tabularnewline
gender & M\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{distribution-of-bin-depth-in-sample}{%
\subsection{Distribution of bin depth in
sample}\label{distribution-of-bin-depth-in-sample}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-5-1.pdf}

\begin{quote}
This graph shows the distribution of the bins according to the depth of
the sequencing. The blue curve represents the distribution of the bins
in the PON. Ideally, the histogram and the blue curve match completely.
If not, this can indicate larger deletions (the first peak decreases) or
larger amplifications (the third peak increases).
\end{quote}

\hypertarget{calculated-numerical-karyotype}{%
\subsection{Calculated numerical
karyotype:}\label{calculated-numerical-karyotype}}

del(1)(q42.13q42.13)

add(1)(q42.2q42.3)

add(2)(p16.3p16.3)

add(2)(q24.1q24.1)

add(3)(p26.3p26.3)

del(3)(p26.2p26.2)

add(3)(q22.3q22.3)

add(3)(q24q24)

del(4)(q21.23q21.23)

add(4)(q23q23)

del(5)(q14.3q33.2)

del(6)(p21.1p11.1)

del(6)(q12q16.3)

add(6)(q25.2q25.2)

del(6)(q26q27)

del(7)(p14.1p11.2)

del(7)(q11.21q36.3)

del(8)(p23.3p23.3)

add(8)(q21.13q21.13)

add(9)(p21.2p21.2)

add(10)(q23.33q23.33)

add(11)(p15.3p11.12)

add(11)(q12.1q25)

del(12)(p13.2p11.23)

add(12)(p11.21p11.21)

del(12)(p11.1p11.1)

del(12)(q13.11q13.11)

del(12)(q21.2q21.31)

del(12)(q21.33q23.2)

add(13)(q22.3q22.3)

add(14)(q12q12)

add(14)(q31.2q31.2)

del(14)(q32.33q32.33)

add(15)(q21.1q21.1)

del(15)(q21.3q21.3)

add(15)(q22.2q22.2)

add(15)(q25.2q25.2)

add(16)(q12.2q13)

del(16)(q24.2q24.2)

del(17)(q21.32q21.32)

add(17)(q22q23.1)

del(18)(p11.31q23)

del(19)(p13.3p13.3)

del(19)(q13.33q13.33)

del(20)(p12.3p12.3)

add(20)(q13.12q13.12)

add(21)(q22.12q22.12)

del(21)(q22.3q22.3)

del(22)(q11.21q11.21)

del(22)(q12.2q12.2)

add(X)(p22.33p11.21)

add(X)(q11.2q26.3)

add(X)(q27.2q28)

del(Y)(p11.31q11.23)

\hypertarget{genes-affected-by-cnvs}{%
\subsection{Genes affected by CNVs*:}\label{genes-affected-by-cnvs}}

\begin{longtable}[]{@{}rrlrlrl@{}}
\toprule
id & sym & bol chr & sta & rt end & abe & rration\tabularnewline
\midrule
\endhead
1604 & ENSG00000123080 & CDKN2C & chr1 & 50960745 & 50974634 &
del\tabularnewline
8082 & ENSG00000121989 & ACVR2A & chr2 & 147844517 & 147930826 &
amp\tabularnewline
8662 & ENSG00000064933 & PMS1 & chr2 & 189784085 & 189877629 &
del\tabularnewline
11346 & ENSG00000181722 & ZBTB20 & chr3 & 114314501 & 115147271 &
del\tabularnewline
13641 & ENSG00000134853 & PDGFRA & chr4 & 54229280 & 54298245 &
del\tabularnewline
13646 & ENSG00000157404 & KIT & chr4 & 54657918 & 54740715 &
del\tabularnewline
14974 & ENSG00000109670 & FBXW7 & chr4 & 152320544 & 152536092 &
del\tabularnewline
16825 & ENSG00000145715 & RASA1 & chr5 & 87267888 & 87391931 &
del\tabularnewline
17123 & ENSG00000134982 & APC & chr5 & 112707498 & 112846239 &
del\tabularnewline
17886 & ENSG00000182578 & CSF1R & chr5 & 150053291 & 150113372 &
del\tabularnewline
17889 & ENSG00000113721 & PDGFRB & chr5 & 150113839 & 150155872 &
del\tabularnewline
19585 & ENSG00000124762 & CDKN1A & chr6 & 36676460 & 36687339 &
del\tabularnewline
21384 & ENSG00000085511 & MAP3K4 & chr6 & 160991727 & 161117385 &
amp\tabularnewline
21681 & ENSG00000122512 & PMS2 & chr7 & 5970925 & 6009106 &
del\tabularnewline
22481 & ENSG00000146648 & EGFR & chr7 & 55019017 & 55211628 &
del\tabularnewline
22928 & ENSG00000263001 & GTF2I & chr7 & 74650231 & 74760692 &
del\tabularnewline
23074 & ENSG00000019991 & HGF & chr7 & 81699010 & 81770438 &
del\tabularnewline
23552 & ENSG00000105851 & PIK3CG & chr7 & 106865278 & 106908980 &
del\tabularnewline
23659 & ENSG00000105976 & MET & chr7 & 116672196 & 116798386 &
del\tabularnewline
23865 & ENSG00000128602 & SMO & chr7 & 129188633 & 129213545 &
del\tabularnewline
24082 & ENSG00000157764 & BRAF & chr7 & 140719327 & 140924928 &
del\tabularnewline
24217 & ENSG00000197993 & KEL & chr7 & 142941114 & 142962363 &
del\tabularnewline
24328 & ENSG00000055130 & CUL1 & chr7 & 148697914 & 148801110 &
del\tabularnewline
24329 & ENSG00000106462 & EZH2 & chr7 & 148807383 & 148884321 &
del\tabularnewline
24436 & ENSG00000106615 & RHEB & chr7 & 151466012 & 151520120 &
del\tabularnewline
24449 & ENSG00000055609 & KMT2C & chr7 & 152134922 & 152436005 &
del\tabularnewline
31516 & ENSG00000097007 & ABL1 & chr9 & 130713946 & 130887675 &
del\tabularnewline
31674 & ENSG00000148400 & NOTCH1 & chr9 & 136494433 & 136546048 &
del\tabularnewline
32366 & ENSG00000133818 & RRAS2 & chr11 & 14277922 & 14364506 &
amp\tabularnewline
32621 & ENSG00000184937 & WT1 & chr11 & 32387775 & 32435885 &
amp\tabularnewline
33142 & ENSG00000198561 & CTNND1 & chr11 & 57753243 & 57819546 &
amp\tabularnewline
33487 & ENSG00000168066 & SF1 & chr11 & 64764606 & 64778786 &
amp\tabularnewline
33490 & ENSG00000133895 & MEN1 & chr11 & 64803510 & 64811294 &
amp\tabularnewline
33764 & ENSG00000110092 & CCND1 & chr11 & 69641087 & 69654474 &
amp\tabularnewline
33855 & ENSG00000165458 & INPPL1 & chr11 & 72223701 & 72239147 &
amp\tabularnewline
34405 & ENSG00000082175 & PGR & chr11 & 101029624 & 101130524 &
amp\tabularnewline
34521 & ENSG00000149311 & ATM & chr11 & 108222484 & 108369102 &
amp\tabularnewline
34754 & ENSG00000118058 & KMT2A & chr11 & 118436464 & 118526832 &
amp\tabularnewline
35859 & ENSG00000165731 & RET & chr10 & 43077069 & 43130351 &
del\tabularnewline
37122 & ENSG00000148737 & TCF7L2 & chr10 & 112950247 & 113167678 &
amp\tabularnewline
37950 & ENSG00000111276 & CDKN1B & chr12 & 12715058 & 12722369 &
del\tabularnewline
37989 & ENSG00000171681 & ATF7IP & chr12 & 14365632 & 14502935 &
del\tabularnewline
38141 & ENSG00000133703 & KRAS & chr12 & 25204789 & 25250936 &
del\tabularnewline
39959 & ENSG00000179295 & PTPN11 & chr12 & 112418351 & 112509913 &
amp\tabularnewline
45273 & ENSG00000140262 & TCF12 & chr15 & 56918623 & 57299281 &
del\tabularnewline
46467 & ENSG00000103126 & AXIN1 & chr16 & 287440 & 352723 &
del\tabularnewline
46622 & ENSG00000103197 & TSC2 & chr16 & 2047967 & 2089491 &
del\tabularnewline
47953 & ENSG00000166164 & BRD7 & chr16 & 50313487 & 50368934 &
amp\tabularnewline
47966 & ENSG00000083799 & CYLD & chr16 & 50742050 & 50801935 &
amp\tabularnewline
48110 & ENSG00000102900 & NUP93 & chr16 & 56730118 & 56850286 &
amp\tabularnewline
48288 & ENSG00000067955 & CBFB & chr16 & 67028984 & 67101058 &
del\tabularnewline
48327 & ENSG00000102974 & CTCF & chr16 & 67562467 & 67639177 &
del\tabularnewline
50883 & ENSG00000120071 & KANSL1 & chr17 & 46029916 & 46225389 &
del\tabularnewline
52042 & ENSG00000141579 & ZNF750 & chr17 & 82829434 & 82840022 &
del\tabularnewline
52137 & ENSG00000177426 & TGIF1 & chr18 & 3411608 & 3459978 &
del\tabularnewline
52741 & ENSG00000152217 & SETBP1 & chr18 & 44680173 & 45068510 &
del\tabularnewline
52799 & ENSG00000175387 & SMAD2 & chr18 & 47808957 & 47931146 &
del\tabularnewline
52876 & ENSG00000141646 & SMAD4 & chr18 & 51028394 & 51085045 &
del\tabularnewline
53051 & ENSG00000171791 & BCL2 & chr18 & 63123346 & 63320128 &
del\tabularnewline
247031 & ENSG00000205542 & TMSB4X & chrX & 12975110 & 12977227 &
amp\tabularnewline
248091 & ENSG00000173674 & EIF1AX & chrX & 20124525 & 20141838 &
amp\tabularnewline
248131 & ENSG00000177189 & RPS6KA3 & chrX & 20149911 & 20267519 &
amp\tabularnewline
249381 & ENSG00000198947 & DMD & chrX & 31097677 & 33339441 &
amp\tabularnewline
250291 & ENSG00000183337 & BCOR & chrX & 40049815 & 40177329 &
amp\tabularnewline
250501 & ENSG00000124486 & USP9X & chrX & 41085445 & 41236579 &
amp\tabularnewline
250551 & ENSG00000215301 & DDX3X & chrX & 41333284 & 41364472 &
amp\tabularnewline
251001 & ENSG00000147050 & KDM6A & chrX & 44873177 & 45112602 &
amp\tabularnewline
251431 & ENSG00000182872 & RBM10 & chrX & 47145221 & 47186813 &
amp\tabularnewline
251601 & ENSG00000078061 & ARAF & chrX & 47561100 & 47571920 &
amp\tabularnewline
253891 & ENSG00000126012 & KDM5C & chrX & 53191321 & 53225422 &
amp\tabularnewline
253981 & ENSG00000072501 & SMC1A & chrX & 53374149 & 53422728 &
amp\tabularnewline
254051 & ENSG00000086758 & HUWE1 & chrX & 53532096 & 53686728 &
amp\tabularnewline
254971 & ENSG00000184675 & AMER1 & chrX & 64185117 & 64205708 &
amp\tabularnewline
255391 & ENSG00000169083 & AR & chrX & 67544021 & 67730619 &
amp\tabularnewline
256071 & ENSG00000184634 & MED12 & chrX & 71118556 & 71142454 &
amp\tabularnewline
256111 & ENSG00000147130 & ZMYM3 & chrX & 71239624 & 71255146 &
amp\tabularnewline
256161 & ENSG00000147133 & TAF1 & chrX & 71366239 & 71532374 &
amp\tabularnewline
257591 & ENSG00000085224 & ATRX & chrX & 77504878 & 77786233 &
amp\tabularnewline
259151 & ENSG00000147202 & DIAPH2 & chrX & 96684663 & 97604997 &
amp\tabularnewline
262571 & ENSG00000174460 & ZCCHC12 & chrX & 118823824 & 118826968 &
amp\tabularnewline
263741 & ENSG00000101972 & STAG2 & chrX & 123960212 & 124422664 &
amp\tabularnewline
264151 & ENSG00000102038 & SMARCA1 & chrX & 129446501 & 129523500 &
amp\tabularnewline
265031 & ENSG00000156531 & PHF6 & chrX & 134373253 & 134428791 &
amp\tabularnewline
268991 & ENSG00000196924 & FLNA & chrX & 154348524 & 154374638 &
amp\tabularnewline
\bottomrule
\end{longtable}

\begin{quote}
Bailey et al. (2018): Comprehensive Characterization of Cancer Driver
Genes and Mutations. Cell 173 (2), 371-385.e18. DOI:
10.1016/j.cell.2018.02.060.;
\end{quote}

\begin{quote}
Papaemmanuil et al. (2016): Genomic Classification and Prognosis in
Acute Myeloid Leukemia. The New England journal of medicine 374 (23), S.
2209--2221. \url{DOI:10.1056/NEJMoa1516192}.
\end{quote}

\hypertarget{karyotype-overview}{%
\subsection{Karyotype overview:}\label{karyotype-overview}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-1.pdf}

\hypertarget{chromosome-plots}{%
\subsection{Chromosome Plots:}\label{chromosome-plots}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-1.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-2.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-3.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-4.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-5.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-6.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-7.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-8.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-9.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-10.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-11.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-12.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-13.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-14.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-15.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-16.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-17.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-18.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-19.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-20.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-21.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-22.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-23.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-12-24.pdf}

\hypertarget{the-legend-contains-two-characteristics}{%
\paragraph{The legend contains two
characteristics:}\label{the-legend-contains-two-characteristics}}

\begin{quote}
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Giemsa stain results. Recognized stain values: gneg, gpos50, gpos75,
  gpos25, gpos100, acen, gvar, stalk.
\item
  The abnormal regions (add or del).
\end{enumerate}
\end{quote}


\end{document}
