\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{09 Dezember, 2020}


\begin{document}
\maketitle

\begin{longtable}[]{@{}ll@{}}
\toprule
\endhead
unique\_reads & 897816\tabularnewline
average\_length & 149.281\tabularnewline
unique\_mapped\_reads & 1749945\tabularnewline
SAMPLE\_ID & 98217.new\tabularnewline
gender & M\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{calculated-numerical-karyotype}{%
\subsection{Calculated numerical
karyotype:}\label{calculated-numerical-karyotype}}

add(1)(q25.2q25.2)

add(3)(p22.3p22.3)

del(3)(p21.32p21.32)

del(3)(p14.3p14.3)

add(3)(q25.31q25.31)

del(5)(q33.2q33.2)

del(7)(q21.11q36.3)

del(9)(q22.1q22.1)

del(11)(q14.2q14.2)

del(12)(q13.2q13.2)

del(16)(p11.2p11.1)

del(16)(q11.2q11.2)

add(18)(q12.2q12.2)

add(21)(q21.3q21.3)

\hypertarget{genes-affected-by-cnvs}{%
\subsection{Genes affected by CNVs*:}\label{genes-affected-by-cnvs}}

\begin{longtable}[]{@{}rrlrlrl@{}}
\toprule
id & sym & bol chr & sta & rt end & abe & rration\tabularnewline
\midrule
\endhead
5025 & ENSG00000069248 & NUP133 & chr1 & 229440259 & 229508341 &
del\tabularnewline
23074 & ENSG00000019991 & HGF & chr7 & 81699010 & 81770438 &
del\tabularnewline
23552 & ENSG00000105851 & PIK3CG & chr7 & 106865278 & 106908980 &
del\tabularnewline
23659 & ENSG00000105976 & MET & chr7 & 116672196 & 116798386 &
del\tabularnewline
23865 & ENSG00000128602 & SMO & chr7 & 129188633 & 129213545 &
del\tabularnewline
24082 & ENSG00000157764 & BRAF & chr7 & 140719327 & 140924928 &
del\tabularnewline
24217 & ENSG00000197993 & KEL & chr7 & 142941114 & 142962363 &
del\tabularnewline
24328 & ENSG00000055130 & CUL1 & chr7 & 148697914 & 148801110 &
del\tabularnewline
24329 & ENSG00000106462 & EZH2 & chr7 & 148807383 & 148884321 &
del\tabularnewline
24436 & ENSG00000106615 & RHEB & chr7 & 151466012 & 151520120 &
del\tabularnewline
24449 & ENSG00000055609 & KMT2C & chr7 & 152134922 & 152436005 &
del\tabularnewline
51230 & ENSG00000108375 & RNF43 & chr17 & 58352500 & 58417595 &
del\tabularnewline
\bottomrule
\end{longtable}

\begin{quote}
Bailey et al. (2018): Comprehensive Characterization of Cancer Driver
Genes and Mutations. Cell 173 (2), 371-385.e18. DOI:
10.1016/j.cell.2018.02.060.;
\end{quote}

\begin{quote}
Papaemmanuil et al. (2016): Genomic Classification and Prognosis in
Acute Myeloid Leukemia. The New England journal of medicine 374 (23), S.
2209--2221. \url{DOI:10.1056/NEJMoa1516192}.
\end{quote}

\hypertarget{karyotype-overview}{%
\subsection{Karyotype overview:}\label{karyotype-overview}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-8-1.pdf}

\hypertarget{chromosome-plots}{%
\subsection{Chromosome Plots:}\label{chromosome-plots}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-1.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-2.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-3.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-4.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-5.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-6.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-7.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-8.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-9.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-10.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-11.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-12.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-13.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-14.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-15.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-16.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-17.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-18.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-19.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-20.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-21.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-22.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-23.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-9-24.pdf}

\hypertarget{the-legend-contains-two-characteristics}{%
\paragraph{The legend contains two
characteristics:}\label{the-legend-contains-two-characteristics}}

\begin{quote}
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Giemsa stain results. Recognized stain values: gneg, gpos50, gpos75,
  gpos25, gpos100, acen, gvar, stalk.
\item
  The abnormal regions (add or del).
\end{enumerate}
\end{quote}


\end{document}
