---
title: "Report CoriandR: ChrOmosomal abeRration Identifier AND Reporter in R"
date: "`r format(Sys.time(), '%d-%m-%Y, %H:%M')`"
output: pdf_document
params:
  data: patient
---

# Calculated Karyotyping for solid tumours

This report documents the calculated karyotyping and estimation of CNVs from sequencing data of solid tumours in research with `coriandR`. 


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
knitr::opts_chunk$set(dev = 'pdf')

library(knitr)
```


```{r sample meta data, echo = FALSE}
# import of sample meta data
patient.meta = data.frame(t(read.table(paste0(as.character(params$data), ".meta.tsv"), stringsAsFactors = F, row.names=1)))
if(tolower(patient.meta$gender) %in% c("f", "female")) {
  patient.meta$gender = "f"
} else {
  patient.meta$gender = "m"
}
```


## Absolute sequencing depth of sequencing data in sample

The following plot visualises the absolute sequencing depth in analysed sample. The grey lines represent the boundaries between the chromosomes. chromosomes, the chromosome names are shown in the lower part of the figure.

```{r data definition, echo = FALSE, fig.width=7, fig.height=3}
# import patient counts from featureCounts output table
patient = read.table(as.character(patient.meta$count_data), header=TRUE, row.names = 1, stringsAsFactors = F)
patient = patient[patient$Chr != 'chrM', ]

# import PON table from coriandR: create a Panel of Normals
# cave: the samples should be obtained and processed under the same conditions (same sample preparation methods like DNA extraction and sequencing technology) as the tumour samples.
pon = read.table(as.character(patient.meta$pon), header=TRUE, row.names = 1, stringsAsFactors = F)

# making PON gender specific for patient (exclude sex chromosomes from opposite gender) and concatenate the data
# for women, only reads on X chromosomes of women are considered for further calculations
if(patient.meta$gender == "f") {
  ponXf = pon[pon$Chr == "chrX_F", ]
  ponXf$Chr = "chrX"
  pon = rbind(pon[pon$gender == "neutral", ], ponXf)
  pon$patient = patient[patient$Chr != "chrY", 6]
} else {
  ponXm = pon[pon$Chr == "chrX_M", ]
  ponXm$Chr = "chrX"
  ponYm = pon[pon$Chr == "chrY_M", ]
  ponYm$Chr = "chrY"
  pon = rbind(pon[pon$gender == "neutral", ], ponXm, ponYm)
  pon$patient = patient[, 6]
}
# the Panel of normals needs recounting after gender segregation
pon$bin = 1:nrow(pon) 

# Calculate normalised patient counts in relation to median sequencing depth of PON times expected ploidy
pon$patient.norm = pon$patient / median(pon$patient)


# import Giemsa-bands for visualisation
# The bands table contains the g-bands of the human genome from
# Cheung VG, Nowak N, Jang W, Kirsch IR, Zhao S, Chen XN, Furey TS, Kim UJ, Kuo WL, Olivier M et al. Integration of cytogenetic landmarks into the draft sequence of the human genome. Nature. 2001 Feb 15;409(6822):953-8. PMID: 11237021
cytobands = read.table("../../tables/cytobands_sort.2.tsv", header=TRUE, sep = "\t", row.names = 1, stringsAsFactors = F)
band_types = data.frame(hue=c(0.3, 1, 1, 1, 1, 1, 0.1, 0.8), saturation=c(1, 0, 0, 0, 0, 0, 1, 1), value=c(1, 1, 0.5, 0.8, 0.7, 0.6, 1, 1), row.names=c("acen", "gneg", "gpos100", "gpos25", "gpos50", "gpos75", "gvar", "stalk"))
cytobands$hue = band_types[cytobands$Band.type, "hue"]
cytobands$saturation = band_types[cytobands$Band.type, "saturation"]
cytobands$value = band_types[cytobands$Band.type, "value"]


# List of relevant genes from German Network for Personalized Medicine
# A list of genes of interest based on the research in germline tumour-detected variants in 49,264 cancer patients (Kuzbari et al. 2023) and genetic dysfunction across all human cancers (Sondka et al. 2018) as well as the database for FDA-recognised human genetic variants `OnkoKB` (Sarah et al. 2024) which can be used in the search for potential targets for the treatment of patients with solid tumours
# Kuzbari Z, Bandlamudi C, Loveday C, Garrett A, Mehine M, George A, Hanson H, Snape K, Kulkarni A, Allen S, Jezdic S, Ferrandino R, Westphalen CB, Castro E, Rodon J, Mateo J, Burghel GJ, Berger MF, Mandelker D, Turnbull C. Germline-focused analysis of tumour-detected variants in 49,264 cancer patients: ESMO Precision Medicine Working Group recommendations. Annals of Oncology. 2023 Mar 1;34(3):215-227. doi: 10.1016/j.annonc.2022.12.003.
# Sondka Z, Bamford S, Cole CG, Ward SA, Dunham I, Forbes SA. The COSMIC Cancer Gene Census: describing genetic dysfunction across all human cancers. Nat Rev Cancer. 2018 Nov;18(11):696-705. doi: 10.1038/s41568-018-0060-1. PMID: 30293088; PMCID: PMC6450507.
# Sarah P. Suehnholz, Moriah H. Nissan, Hongxin Zhang, Ritika Kundra, Subhiksha Nandakumar, Calvin Lu, Stephanie Carrero, Amanda Dhaneshwar, Nicole Fernandez, Benjamin W. Xu, Maria E. Arcila, Ahmet Zehir, Aijazuddin Syed, A. Rose Brannon, Julia E. Rudolph, Eder Paraiso, Paul J. Sabbatini, Ross L. Levine, Ahmet Dogan, Jianjiong Gao, Marc Ladanyi, Alexander Drilon, Michael F. Berger, David B. Solit, Nikolaus Schultz, Debyani Chakravarty; Quantifying the Expanding Landscape of Clinical Actionability for Patients with Cancer. Cancer Discov 1 January 2024; 14 (1): 49–65. https://doi.org/10.1158/2159-8290.CD-23-0467
genes_presision_medicine = read.table("../../tables/genes_presision_medicine.csv", 
                                      header=TRUE, sep = ";", stringsAsFactors = F)

# List of all genes in human genome
# genes of interest will be picked out with their coordinates if located in regions with amplification/deletion
all_genes = read.table("../../tables/all_genes.tsv", header=TRUE, row.names = 1, stringsAsFactors = F)
all_genes$start = as.numeric(all_genes$start)
all_genes$end = as.numeric(all_genes$end)

# List of chromosomes for plotting
contigs = data.frame(pos=tapply(pon$bin, pon$Chr, mean))
contigs$chr = rownames(contigs)
contigs$Name = gsub("chr", "", rownames(contigs))
contigs = contigs[order(as.numeric(contigs$Name)),]
# definition of coordinates for chromosomal p- and q-arms
contigs$p_start = sapply(contigs$chr, function(c) min(c(cytobands$Start[cytobands$Chr == c & grepl('p', cytobands$Name)])))
contigs$p_end = sapply(contigs$chr, function(c) max(c(cytobands$End[cytobands$Chr == c & grepl('p', cytobands$Name)])))
contigs$q_start = sapply(contigs$chr, function(c) min(c(cytobands$Start[cytobands$Chr == c & grepl('q', cytobands$Name)])))
contigs$q_end = sapply(contigs$chr, function(c) max(c(cytobands$End[cytobands$Chr == c & grepl('q', cytobands$Name)])))

# assignment of giemsa-bands to coordinates of chromosomal p- and q-arms
pon$arm = sapply(pon$bin, function(b) {
  ifelse(contigs$p_start[contigs$chr == pon$Chr[b]] <= pon$Start[b] &
           pon$End[b] <= contigs$p_end[contigs$chr == pon$Chr[b]], 'p',
         ifelse(contigs$q_start[contigs$chr == pon$Chr[b]] <= pon$Start[b] &
                  pon$End[b] <= contigs$q_end[contigs$chr == pon$Chr[b]], 'q', NA))
  })

# a plot with sequencing depth of tumour data
par(mar=c(2, 4, 1, 1))
plot(pon$bin, pon$patient, ylim = c(0, qnorm(0.999, mean(pon$patient), sd(pon$patient))),
     col = "#11111111",
     pch = 16, xaxt="n", xlab="", las = 1, ylab="Read depth", xaxs="i")
abline(v = c(pon$bin[pon$Start == 1], nrow(pon)), col="#55555555", cex = 0.5)
with(contigs[(1:(nrow(contigs)/2))*2, ], axis(side=1, labels=Name, at=pos, line=-.5, cex.axis=0.5, tick=F))
with(contigs[(1:(nrow(contigs)/2))*2-1, ], axis(side=1, labels=Name, at=pos, line=-1, cex.axis=0.5, tick=F))
```


## Sample sequencing characteristics and mapping statistics

```{r mapping statistics, echo = FALSE}
# file 'stats' was created with a bash script from the aligned sequences and contains the information about
# the number of raw read pairs, average read length and unique mapping pairs
# parameters 'name' and 'gender' come from samples metafile
stats = data.frame(t(read.table(as.character(patient.meta$mapping_stats), stringsAsFactors = F, row.names=1)))
stats$unique_mapping_pairs = round(sum(patient[ , 6]/2))

stats$raw_read_pairs = format(stats$raw_read_pairs, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
stats$average_read_length = format(stats$average_read_length, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
stats$unique_mapping_pairs = format(stats$unique_mapping_pairs, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)

```

| Sample characterisic | Value |
|---------------------:|:------|
| **Sample name**      | **`r as.character(patient.meta$name)`**     |
| **Sample gender**    | **`r as.character(patient.meta$gender)`**   |
| Raw read pairs       | `r as.character(stats$raw_read_pairs)`      |
| Average read length  | `r as.character(stats$average_read_length)` |
| Unique mapping pairs | `r as.character(stats$unique_mapping_pairs)`|


## Distribution of sequencing depth per bin in sample

This picture shows the distribution of bins (with density on the y-axis) according to the number of reads per bin (x-axis). The red curve represents the distribution of the PON scaled to the same depth.

Ideally, the histogram and the red curve match completely. If not, this can indicate larger deletions (the first peak is smaller then the PON) or larger amplifications (the third peak becomes higher). The labels `x1`, `x2`, `x3`, `x4` indicate the ploidy of the bins. 

```{r echo = FALSE, fig.width=7, fig.height=4}
with(pon[!pon$blocked, ], 
     {
      hist(patient, breaks=seq(0, max(patient)+20, by=20),
           xlim=c(0, median(patient)*2),
           freq=F, xlab="Depth per bin", border=T, col="blue", main="")
      pon.hist = hist(norm.clean.median*median(patient), breaks=seq(0, max(norm.clean.median*median(patient)*2,  na.rm=T), by=20), plot=F)
      lines(pon.hist$mids, pon.hist$density, col="red")
      abline(v = c(0.5, 1, 1.5, 2) *median(patient), col='gray', lty=2)
      legend("topright", legend="scaled PON", col="red", lty=1, cex=0.8)
      mtext(c("x1", "x2", "x3", "x4"), at=c(0.5, 1, 1.5, 2) *median(patient), side=3)
    })
```

\newpage

```{r gauss-test, echo = FALSE, fig.width=8, fig.height=3}
sd = mean(unlist(pon$norm.clean.sd), na.rm=T)
pon$pat.z = (pon$patient.norm-1) / sd
pat.dens = density(pon$pat.z, na.rm=T)

# Presentation of patient Z-values compared to estimated PON distribution
plot(seq(from=-20, to=20, by=0.1), dnorm(seq(from=-20, to=20, by=0.1)), type='l', col="gray",
     main="Patient Z-values compared to estimated PON distribution", ylab="Density", xlab="Z-value", ylim=c(0, max(pat.dens$y)))
lines(pat.dens, col="black")
legend("topright", legend=c("theoretical Z-Distribution", "Patient"), col=c("gray", "black"), lty=1, cex=0.7)


# Detection of all bins that are outside the normal distribution after applying a Gauss test and after p-value adjustment according to the Benjamini-Hochberg method 
pon$gauss = apply(pon[, c("patient.norm", "norm.clean.median", "norm.clean.sd")], 1, function(x) ifelse(x[1] == 0, 1, min(pnorm(q=x[1], mean=x[2], sd=x[3], lower.tail=x[1]<x[2])*2, 1)))
pon$gauss.adjust = p.adjust(pon$gauss, method="hochberg")
pon$gauss.adjust[is.na(pon$gauss.adjust)] = 1
par(mar=c(2, 4, 1, 1))
with(pon[!pon$blocked, ],
     plot(bin, ifelse(patient.norm > 2, 4, patient.norm*2), ylim=c(0, 4),
          col = ifelse(gauss.adjust < 0.05, "#ff000033", "#11111111"),
          pch = ifelse(patient.norm > 2, 17, 20), xaxt="n", xlab="",
          main="Overview of the bins deviating from the normal distribution", las = 1, ylab="Ploidy", xaxs="i")
)

abline(v = c(pon$bin[pon$Start == 1], nrow(pon)), col="#55555555", cex = 0.5)
with(contigs[(1:(nrow(contigs)/2))*2, ], axis(side=1, labels=Name, at=pos, line=-.5, cex.axis=0.5, tick=F))
with(contigs[(1:(nrow(contigs)/2))*2-1, ], axis(side=1, labels=Name, at=pos, line=-1, cex.axis=0.5, tick=F))
```

In the first step of calculated karyotyping, the sequencing data of a tumour sample are normalized by the median sequencing depth per bin. Thereafter, we used standardisation with calculation of the pseudo z-values of the distribution of the bins. A z-score represents the number of standard deviations from the standard value of the reference population (PON) for the analysed sample. The first picture visualises the z-values of the sample and PON.

In addition, we tested pseudo z-scores against a normal distribution with parameters of the PON in a two-tailed test. The obtained p-values were adjusted using the Benjamini-Hochberg method (Benjamini and Hochberg 1995) in control of the false discovery rate. In the second picture, an overview of normalised bins (data points) with expectes ploidy (y-axis) is shown across the chromosomal coordinates (x-axis). Bins with the read distribution outside the 99 % of the normal distribution are marked in red.


```{r high level abberations in bins, echo = FALSE}
# Definition of high level abberant bins in tumour sample since they are better targets for cancer therapy
# A deletion in a bin is detected if the normalized value for that bin is below 0.5 * median for 
# all normalized PON samples with a significance level of alpha = 0.05. 
# An amplification leads to a value above 2.5 * median with a significance level of alpha = 0.05. 
pon$aberration = factor(ifelse(pon$patient.norm > 2.5*pon$norm.clean.median & pon$gauss.adjust < 0.05, 'amp', ifelse(pon$patient.norm < 0.5*pon$norm.clean.median & pon$gauss.adjust < 0.05, 'del', 'normal')), levels = c('normal', 'amp', 'del'))
```

```{r gc content correlation, echo = FALSE}
# correlation plot of normalised median in sample / in PON to gc content
plot(pon$patient.norm/pon$norm.clean.median, pon$gc, type='p',
     main="Correlation ratio patient/median PON to gc content", pch = 16,
     xlab="Patient/median(PON)", ylab="GC-Content", col="#11111111")
```

This picture displays the correlation of normalised median in sample / in PON to gc content in human reference genome. The gc content should be between 35 % and 60 %. High correlation between normalised median in sample / in PON to gc content can be a result of large abberations or sequencing bias.

\newpage

## Calculated numerical karyotype:

```{r transformation of bins to giemsa bands and create of digital karyotype, echo = FALSE}
cytobands$aberration = factor(apply(cytobands, 1, function(x) {
  bins = pon[pon$Chr == x[1] & pon$Start >= floor(as.integer(x[2])/10^6)*10^6 & pon$End <= ceiling(as.integer(x[3])/10^6)*10^6 & !pon$blocked,]
  if(nrow(bins) < 1) return("normal")
  if(sum(bins$aberration == "del")/nrow(bins) > 0.5) return("del")
  if(sum(bins$aberration == "amp")/nrow(bins) > 0.5) return("amp")
  return("normal")
}))

cytobands$hue[cytobands$aberration == "amp"] = 0.65
cytobands$saturation[cytobands$aberration == "amp"] = 1
cytobands$hue[cytobands$aberration == "del"] = 0
cytobands$saturation[cytobands$aberration == "del"] = 1

# bands merging
abnorm.bands = cytobands[cytobands$aberration != "normal", ]
colnames(abnorm.bands)[colnames(abnorm.bands) == "Name"] = "Start.band"
karyotype_to_report = abnorm.bands[1, ]

if (nrow(abnorm.bands) >= 2) {
  k = 1
  # the counter in table karyotype_to_report
  for (s in 2:nrow(abnorm.bands)) {
    if (karyotype_to_report$Chr[k] == abnorm.bands$Chr[s] &
        karyotype_to_report$aberration[k] == abnorm.bands$aberration[s] &
        karyotype_to_report$End[k] == abnorm.bands$Start[s]) 
      { 
        karyotype_to_report$End.band[k] = abnorm.bands$Start.band[s]
        karyotype_to_report$End[k] = abnorm.bands$End[s]
        k = nrow(karyotype_to_report)
    } else {
       karyotype_to_report[k+1, ] = abnorm.bands[s, ]
       karyotype_to_report[k+1, "End.band"] = karyotype_to_report[k+1, "Start.band"]
       # add the band on a new chr or with a different aberration
       k = nrow(karyotype_to_report)
       # this is now the counter band
    }
  }
  for (g in 1:nrow(karyotype_to_report)) {
    if (is.na(karyotype_to_report$End.band[g])) {
      karyotype_to_report$End.band[g] <- as.character(karyotype_to_report$Start.band[g])
    }
  }
} 

karyotype_to_report$karyotype = as.character(paste0(karyotype_to_report$aberration, "(",
                                                       sub("chr", "", karyotype_to_report$Chr), ")(",
                                                       karyotype_to_report$Start.band,
                                                       karyotype_to_report$End.band, ")"))

```

```{r karyotyping of whole chromosome arms and numerical chromosome changes}
for(c in 1:nrow(contigs)){
  if(sum(pon$aberration[pon$Chr == contigs$chr[c]] == 'amp')/sum(pon$Chr == contigs$chr[c]) > 0.8) karyotype_to_report[(nrow(karyotype_to_report) +1), "karyotype"] = as.character(paste('$+$', c))
  else if(sum(pon$aberration[pon$Chr == contigs$chr[c]] == 'del')/sum(pon$Chr == contigs$chr[c]) > 0.8) karyotype_to_report[(nrow(karyotype_to_report) +1), "karyotype"] = as.character(paste('$-$', c))
  # numerical chromosome changes
  
  else if(sum(pon$aberration[pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'p'] == 'amp')/sum(pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'p') > 0.8) karyotype_to_report[(nrow(karyotype_to_report) +1), "karyotype"] = as.character(paste('$+$', c, 'p'))
  else if(sum(pon$aberration[pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'p'] == 'del')/sum(pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'p') > 0.8) karyotype_to_report[(nrow(karyotype_to_report) +1), "karyotype"] = as.character(paste('$-$', c, 'p'))
  else if(sum(pon$aberration[pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'q'] == 'amp')/sum(pon$Chr == contigs$chr[c]& !is.na(pon$arm) & pon$arm == 'q') > 0.8) karyotype_to_report[(nrow(karyotype_to_report) +1), "karyotype"] = as.character(paste('$+$', c, 'q'))
  else if(sum(pon$aberration[pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'q'] == 'del')/sum(pon$Chr == contigs$chr[c] & !is.na(pon$arm) & pon$arm == 'q') > 0.8) karyotype_to_report[(nrow(karyotype_to_report) +1), "karyotype"] = as.character(paste('$-$', c, 'q'))
    # whole chromosome arms is del or add
}


kable(karyotype_to_report$karyotype, col.names = NULL, align = "l", "html")
```

The presentation of the calculated numerical karyotype is based on the ISCN nomenclature.

International Standing Committee on Human Cytogenomic Nomenclature and McGowan-Jordan, J. and Hastings, R.J. and Moore, S. (2020): ISCN 2020: An International System for Human Cytogenomic Nomenclature. In: An International system for human cytogenetic nomenclature. Karger Publishers. ISBN: 9783318067064.

\newpage

## Genes affected by CNVs (copy number variations)\*:

```{r genes in aberration bins table, echo = FALSE}
  special_genes = all_genes[all_genes$symbol %in% genes_presision_medicine$Gene, ]
  special_genes$aberration = 'norm'
  special_genes$found = 0
  
  for(i in 1:nrow(special_genes)) {
    genes.in.bin = pon[(pon$aberration != "normal" & pon$Chr == special_genes$chr[i] & 
                          pon$Start <= special_genes$start[i] & 
                          special_genes$start[i] <= pon$End) |
                       (pon$aberration != "normal" & 
                        pon$Chr == special_genes$chr[i] & 
                        pon$Start <= special_genes$end[i] & 
                        special_genes$end[i] <= pon$End), ]
    
    if(nrow(genes.in.bin) == 0) {next}
    special_genes$found[i] = 1
    if(all(genes.in.bin$aberration == 'amp')) {special_genes$aberration[i] = 'amp'}
    if(all(genes.in.bin$aberration == 'del')) {special_genes$aberration[i] = 'del'}
  }
  
reported_genes = special_genes[special_genes$found == 1 & (special_genes$aberration == "amp" | special_genes$aberration == "del"), ]
reported_genes$start = format(reported_genes$start, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
reported_genes$end = format(reported_genes$end, decimal.mark = ",", big.mark = ".", small.mark = " ", small.interval = 3)
  
kable(reported_genes[ , 1:6], align = "lllrrl")
```

Bailey et al. (2018): Comprehensive Characterization of Cancer Driver Genes and Mutations. Cell 173 (2), 371-385.e18. DOI: 10.1016/j.cell.2018.02.060.;

Papaemmanuil et al. (2016): Genomic Classification and Prognosis in Acute Myeloid Leukemia. The New England journal of medicine 374 (23), S. 2209–2221. DOI:10.1056/NEJMoa1516192.

\newpage

## Chromosome overview plots:

```{r simple chromosomal plots, echo = FALSE, fig.width=12, fig.height=4}
#units = c('0', '00', 'K', '0K', '00K', 'M', '0M', '00M', 'B', 'B', 'B', 'T', 'T', 'T')
par(mar=c(4, 4, 0.5, 5.5))

for(i in unique(pon$Chr)) {
  with(pon[pon$Chr == i,], {
    plot(Start, ifelse(patient.norm > 2.5, 5, patient.norm*2), 
         col = ifelse(blocked, 0, c("black", "blue", "red")[aberration]), 
         ylim=c(0,5), xlim=c(0,max(End)), xaxt='n', xaxs="i", xlab='', ylab='Ploidy', 
         pch=ifelse(patient.norm > 2.5, 17, 20), las=2)
    segments(Start, (norm.clean.median+norm.clean.iqr)*2, Start, (norm.clean.median-norm.clean.iqr)*2, col="#aaaaaaaa")
    #mag = floor(log10(max(End)))
    #xmax = floor(max(End)/10^mag+1)*10^mag
    axis(1, seq(0, 10^9, by=2*10^7), paste(seq(0, 1000, by=20), "Mbp"), line=2, tck=-0.05)
    axis(1, seq(0, 10^9, by=10^7), rep("", 10^9/10^7+1), line=2)
    #axis(1, seq(0, mag+2)*10^(mag), c('0 bp', paste0(seq(1, mag+2), units[mag], ' bp')), line=2)
  })
  mtext(i, side=2, line=0.9, at=-0.4, las=2, cex=1.5)
  abline(h=0:4, lty=3, col="#aaaaaaaa")
  with(cytobands[cytobands$Chr == i, ], {
    rect(Start, -.25, End, -.55, xpd=T,
         col=hsv(hue, saturation, value, 0.3)
         )
  })
  legend("right", c("amplification", "gain*", "deletion", "stalk", "centromer"), fill=c("white", "white", "white", "#aaaa0055", "#00990055"), col=c("blue", "blue", "red", NA, NA), xpd=T, inset=c(-.1, 0), border=F, lty=0, pch=c(20, 2, 20, NA, NA), cex=0.7)
}
```

\newpage

\* The reads that are marked with triangles have a particularly deep sequencing depth, i.e. there are more than 4 copies.

\*\* The legend contains characteristics to Giemsa stain results (Cheung 2001). Recognised stain values: gneg, gpos50, gpos75, gpos25, gpos100, acen, gvar, stalk.

Cheung VG, Nowak N, Jang W, Kirsch IR, Zhao S, Chen XN, Furey TS, Kim UJ, Kuo WL, Olivier M et al. Integration of cytogenetic landmarks into the draft sequence of the human genome. Nature. 2001 Feb 15;409(6822):953-8. PMID: 11237021



