\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Report CORIANDR: ChrOmosomal abeRration Identifier AND Reporter in R}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{25 Februar, 2021}


\begin{document}
\maketitle

\hypertarget{sample-id}{%
\subsection{Sample ID}\label{sample-id}}

\begin{verbatim}
## [1] "98217.new.report"
\end{verbatim}

\begin{longtable}[]{@{}ll@{}}
\toprule
\endhead
row\_read\_pairs & 897816\tabularnewline
average\_read\_length & 149.281\tabularnewline
unique\_mapped\_pairs & 874972.5\tabularnewline
gender & M\tabularnewline
\bottomrule
\end{longtable}

\#\#Bins distribution
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-5-1.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-5-2.pdf}

\hypertarget{calculated-numerical-karyotype}{%
\subsection{Calculated numerical
karyotype:}\label{calculated-numerical-karyotype}}

del(1)(p36.31p36.31)

del(1)(p36.12p36.11)

del(1)(p35.1p35.1)

del(1)(p34.2p34.2)

del(1)(p31.3p31.3)

del(1)(q24.1q24.1)

add(1)(q31.1q31.2)

del(1)(q42.12q42.2)

del(2)(p25.1p25.1)

del(2)(p22.2p22.2)

add(2)(p16.3p16.3)

add(2)(p16.1p16.1)

add(2)(p11.2p11.2)

del(2)(q21.2q21.2)

add(2)(q32.1q32.1)

del(2)(q32.2q32.2)

add(2)(q34q34)

del(2)(q36.1q36.1)

add(3)(p26.3p26.3)

del(3)(p26.2p26.2)

del(3)(p25.2p25.1)

add(3)(p24.3p24.3)

del(3)(p23p23)

add(3)(p22.3p22.3)

del(3)(p21.32p21.32)

del(3)(p14.3p14.3)

del(3)(p14.1p14.1)

add(3)(p12.2p12.2)

add(3)(p11.2p11.2)

add(3)(q11.1q11.2)

add(3)(q13.2q13.2)

del(3)(q21.1q21.1)

add(3)(q24q24)

add(3)(q25.31q25.31)

del(3)(q25.32q25.32)

add(3)(q26.1q26.1)

del(3)(q27.2q27.2)

del(4)(p16.1p16.1)

add(4)(p12p12)

add(4)(q13.1q13.1)

del(4)(q21.3q21.3)

add(4)(q22.2q22.2)

add(4)(q28.2q28.2)

del(4)(q31.1q31.1)

add(4)(q34.3q34.3)

del(5)(p13.1p13.1)

del(5)(q14.1q14.1)

add(5)(q14.3q21.2)

add(5)(q22.1q22.1)

del(5)(q33.2q33.3)

del(5)(q35.1q35.2)

del(6)(p21.33p21.33)

del(6)(p21.1p21.1)

del(6)(p11.2p11.2)

add(6)(q13q14.1)

add(6)(q16.1q16.1)

add(6)(q22.1q22.1)

add(6)(q22.33q22.33)

del(6)(q23.2q23.3)

add(6)(q25.2q25.2)

del(6)(q25.3q25.3)

add(7)(p12.2p12.1)

del(7)(q11.21q11.21)

del(7)(q21.11q36.3)

del(8)(p21.3p21.3)

del(8)(p21.1p12)

add(8)(q12.3q12.3)

add(8)(q21.12q21.13)

del(9)(p24.1p24.1)

add(9)(p21.2p21.2)

del(9)(q21.11q21.11)

del(9)(q33.1q33.1)

del(9)(q33.3q33.3)

add(10)(p11.1p11.1)

add(10)(q21.1q21.1)

del(10)(q22.1q22.1)

del(10)(q26.11q26.11)

del(10)(q26.13q26.13)

del(11)(p15.2p15.1)

add(11)(p14.2p14.1)

del(11)(p11.2p11.12)

del(11)(q13.4q13.4)

del(11)(q14.2q14.2)

del(11)(q23.2q23.2)

add(12)(p12.3p12.3)

add(12)(p12.1p12.1)

del(12)(q13.12q13.12)

del(12)(q13.2q13.3)

add(12)(q21.32q21.32)

del(12)(q23.3q23.3)

del(12)(q24.12q24.12)

del(12)(q24.23q24.23)

add(13)(q11q11)

del(13)(q12.12q12.12)

del(13)(q14.11q14.11)

add(13)(q21.33q21.33)

add(13)(q22.3q22.3)

add(13)(q33.2q33.2)

add(14)(q12q12)

add(14)(q21.2q21.2)

del(14)(q23.1q23.1)

del(14)(q24.2q24.2)

add(14)(q31.2q31.2)

del(14)(q32.11q32.11)

add(15)(q21.1q21.1)

del(15)(q21.2q21.3)

add(15)(q22.2q22.2)

add(15)(q25.2q25.2)

del(15)(q26.1q26.1)

del(16)(p13.11p12.3)

del(16)(p11.1p11.1)

del(16)(q12.2q12.2)

add(16)(q13q13)

del(16)(q22.1q22.1)

add(16)(q22.3q22.3)

del(17)(p13.3p13.3)

add(17)(p11.1p11.1)

del(17)(q12q12)

del(17)(q21.32q21.32)

del(17)(q22q22)

del(18)(q11.2q11.2)

add(18)(q12.2q12.2)

add(18)(q21.2q21.2)

add(18)(q22.2q22.3)

del(19)(p13.12p13.12)

del(19)(q13.11q13.11)

del(19)(q13.33q13.41)

add(20)(p12.1p12.1)

del(20)(p11.21p11.21)

del(20)(q13.13q13.13)

add(21)(q21.3q21.3)

add(21)(q22.13q22.13)

add(22)(q11.21q11.21)

del(22)(q12.1q12.2)

del(22)(q13.1q13.1)

add(22)(q13.33q13.33)

add(X)(p22.33q28)

del(Y)(p11.31q12)

\hypertarget{genes-affected-by-cnvs}{%
\subsection{Genes affected by CNVs*:}\label{genes-affected-by-cnvs}}

\begin{longtable}[]{@{}rrlrlrl@{}}
\toprule
id & sym & bol chr & sta & rt end & abe & rration\tabularnewline
\midrule
\endhead
882 & ENSG00000117713 & ARID1A & chr1 & 26693236 & 26782104 &
del\tabularnewline
1201 & ENSG00000163874 & ZC3H12A & chr1 & 37474580 & 37484377 &
del\tabularnewline
1604 & ENSG00000123080 & CDKN2C & chr1 & 50960745 & 50974634 &
amp\tabularnewline
2967 & ENSG00000265972 & TXNIP & chr1 & 145992435 & 145996600 &
del\tabularnewline
4392 & ENSG00000163435 & ELF3 & chr1 & 202007945 & 202017183 &
del\tabularnewline
4895 & ENSG00000163041 & H3F3A & chr1 & 226061851 & 226072007 &
del\tabularnewline
5025 & ENSG00000069248 & NUP133 & chr1 & 229440259 & 229508341 &
del\tabularnewline
5860 & ENSG00000119772 & DNMT3A & chr2 & 25227855 & 25342590 &
del\tabularnewline
5866 & ENSG00000143970 & ASXL2 & chr2 & 25733753 & 25878516 &
del\tabularnewline
6183 & ENSG00000152518 & ZFP36L2 & chr2 & 43222402 & 43226606 &
del\tabularnewline
6626 & ENSG00000169564 & PCBP1 & chr2 & 70087454 & 70089203 &
del\tabularnewline
8082 & ENSG00000121989 & ACVR2A & chr2 & 147844517 & 147930826 &
del\tabularnewline
8662 & ENSG00000064933 & PMS1 & chr2 & 189784085 & 189877629 &
del\tabularnewline
9038 & ENSG00000178568 & ERBB4 & chr2 & 211375717 & 212538841 &
amp\tabularnewline
10405 & ENSG00000181555 & SETD2 & chr3 & 47016429 & 47163967 &
del\tabularnewline
11043 & ENSG00000044524 & EPHA3 & chr3 & 89107621 & 89482134 &
amp\tabularnewline
11346 & ENSG00000181722 & ZBTB20 & chr3 & 114314501 & 115147271 &
amp\tabularnewline
13641 & ENSG00000134853 & PDGFRA & chr4 & 54229280 & 54298245 &
del\tabularnewline
13646 & ENSG00000157404 & KIT & chr4 & 54657918 & 54740715 &
del\tabularnewline
14974 & ENSG00000109670 & FBXW7 & chr4 & 152320544 & 152536092 &
amp\tabularnewline
16825 & ENSG00000145715 & RASA1 & chr5 & 87267888 & 87391931 &
amp\tabularnewline
18512 & ENSG00000164379 & FOXQ1 & chr6 & 1312098 & 1314758 &
del\tabularnewline
19285 & ENSG00000234745 & HLA-B & chr6 & 31269491 & 31357188 &
del\tabularnewline
21384 & ENSG00000085511 & MAP3K4 & chr6 & 160991727 & 161117385 &
del\tabularnewline
21626 & ENSG00000198286 & CARD11 & chr7 & 2906141 & 3043945 &
del\tabularnewline
21681 & ENSG00000122512 & PMS2 & chr7 & 5970925 & 6009106 &
del\tabularnewline
23074 & ENSG00000019991 & HGF & chr7 & 81699010 & 81770438 &
del\tabularnewline
23552 & ENSG00000105851 & PIK3CG & chr7 & 106865278 & 106908980 &
del\tabularnewline
23659 & ENSG00000105976 & MET & chr7 & 116672196 & 116798386 &
del\tabularnewline
23865 & ENSG00000128602 & SMO & chr7 & 129188633 & 129213545 &
del\tabularnewline
24082 & ENSG00000157764 & BRAF & chr7 & 140719327 & 140924928 &
del\tabularnewline
24217 & ENSG00000197993 & KEL & chr7 & 142941114 & 142962363 &
del\tabularnewline
24328 & ENSG00000055130 & CUL1 & chr7 & 148697914 & 148801110 &
del\tabularnewline
24329 & ENSG00000106462 & EZH2 & chr7 & 148807383 & 148884321 &
del\tabularnewline
24436 & ENSG00000106615 & RHEB & chr7 & 151466012 & 151520120 &
del\tabularnewline
24449 & ENSG00000055609 & KMT2C & chr7 & 152134922 & 152436005 &
del\tabularnewline
27499 & ENSG00000179388 & EGR3 & chr8 & 22687659 & 22693480 &
del\tabularnewline
28554 & ENSG00000176571 & CNBD1 & chr8 & 86866442 & 87615219 &
amp\tabularnewline
29781 & ENSG00000147889 & CDKN2A & chr9 & 21967753 & 21995301 &
amp\tabularnewline
31516 & ENSG00000097007 & ABL1 & chr9 & 130713946 & 130887675 &
del\tabularnewline
32621 & ENSG00000184937 & WT1 & chr11 & 32387775 & 32435885 &
amp\tabularnewline
36150 & ENSG00000150347 & ARID5B & chr10 & 61901684 & 62096944 &
amp\tabularnewline
37122 & ENSG00000148737 & TCF7L2 & chr10 & 112950247 & 113167678 &
del\tabularnewline
37379 & ENSG00000170430 & MGMT & chr10 & 129467190 & 129770983 &
amp\tabularnewline
38141 & ENSG00000133703 & KRAS & chr12 & 25204789 & 25250936 &
amp\tabularnewline
38282 & ENSG00000188375 & H3F3C & chr12 & 31791185 & 31792298 &
del\tabularnewline
38444 & ENSG00000189079 & ARID2 & chr12 & 45729706 & 45908040 &
amp\tabularnewline
38568 & ENSG00000167548 & KMT2D & chr12 & 49018975 & 49059774 &
del\tabularnewline
38922 & ENSG00000065361 & ERBB3 & chr12 & 56076799 & 56103505 &
del\tabularnewline
40154 & ENSG00000135100 & HNF1A & chr12 & 120978543 & 121002512 &
del\tabularnewline
40605 & ENSG00000121741 & ZMYM2 & chr13 & 19958670 & 20091829 &
del\tabularnewline
40632 & ENSG00000150457 & LATS2 & chr13 & 20973036 & 21061586 &
del\tabularnewline
41388 & ENSG00000276644 & DACH1 & chr13 & 71437966 & 71867204 &
amp\tabularnewline
42295 & ENSG00000129474 & AJUBA & chr14 & 22971177 & 22982551 &
amp\tabularnewline
45273 & ENSG00000140262 & TCF12 & chr15 & 56918623 & 57299281 &
del\tabularnewline
46622 & ENSG00000103197 & TSC2 & chr16 & 2047967 & 2089491 &
amp\tabularnewline
48110 & ENSG00000102900 & NUP93 & chr16 & 56730118 & 56850286 &
amp\tabularnewline
48288 & ENSG00000067955 & CBFB & chr16 & 67028984 & 67101058 &
del\tabularnewline
48327 & ENSG00000102974 & CTCF & chr16 & 67562467 & 67639177 &
del\tabularnewline
48539 & ENSG00000140836 & ZFHX3 & chr16 & 72782885 & 73891871 &
amp\tabularnewline
50479 & ENSG00000167258 & CDK12 & chr17 & 39461486 & 39564907 &
del\tabularnewline
50489 & ENSG00000141736 & ERBB2 & chr17 & 39687914 & 39730426 &
del\tabularnewline
50883 & ENSG00000120071 & KANSL1 & chr17 & 46029916 & 46225389 &
del\tabularnewline
51230 & ENSG00000108375 & RNF43 & chr17 & 58352500 & 58417595 &
del\tabularnewline
51776 & ENSG00000161547 & SRSF2 & chr17 & 76734115 & 76737374 &
del\tabularnewline
53051 & ENSG00000171791 & BCL2 & chr18 & 63123346 & 63320128 &
del\tabularnewline
55117 & ENSG00000125726 & CD70 & chr19 & 6583183 & 6604103 &
amp\tabularnewline
56102 & ENSG00000105173 & CCNE1 & chr19 & 29811991 & 29824312 &
del\tabularnewline
56943 & ENSG00000105464 & GRIN2D & chr19 & 48394875 & 48444931 &
del\tabularnewline
58601 & ENSG00000100030 & MAPK1 & chr22 & 21754500 & 21867680 &
amp\tabularnewline
59615 & ENSG00000196576 & PLXNB2 & chr22 & 50274979 & 50307646 &
amp\tabularnewline
60103 & ENSG00000156304 & SCAF4 & chr21 & 31671033 & 31732075 &
del\tabularnewline
247031 & ENSG00000205542 & TMSB4X & chrX & 12975110 & 12977227 &
amp\tabularnewline
248091 & ENSG00000173674 & EIF1AX & chrX & 20124525 & 20141838 &
amp\tabularnewline
248131 & ENSG00000177189 & RPS6KA3 & chrX & 20149911 & 20267519 &
amp\tabularnewline
249381 & ENSG00000198947 & DMD & chrX & 31097677 & 33339441 &
amp\tabularnewline
250291 & ENSG00000183337 & BCOR & chrX & 40049815 & 40177329 &
amp\tabularnewline
250501 & ENSG00000124486 & USP9X & chrX & 41085445 & 41236579 &
amp\tabularnewline
250551 & ENSG00000215301 & DDX3X & chrX & 41333284 & 41364472 &
amp\tabularnewline
251001 & ENSG00000147050 & KDM6A & chrX & 44873177 & 45112602 &
amp\tabularnewline
251431 & ENSG00000182872 & RBM10 & chrX & 47145221 & 47186813 &
amp\tabularnewline
251601 & ENSG00000078061 & ARAF & chrX & 47561100 & 47571920 &
amp\tabularnewline
253891 & ENSG00000126012 & KDM5C & chrX & 53191321 & 53225422 &
amp\tabularnewline
253981 & ENSG00000072501 & SMC1A & chrX & 53374149 & 53422728 &
amp\tabularnewline
254051 & ENSG00000086758 & HUWE1 & chrX & 53532096 & 53686728 &
amp\tabularnewline
254971 & ENSG00000184675 & AMER1 & chrX & 64185117 & 64205708 &
amp\tabularnewline
255391 & ENSG00000169083 & AR & chrX & 67544021 & 67730619 &
amp\tabularnewline
256071 & ENSG00000184634 & MED12 & chrX & 71118556 & 71142454 &
amp\tabularnewline
256111 & ENSG00000147130 & ZMYM3 & chrX & 71239624 & 71255146 &
amp\tabularnewline
256161 & ENSG00000147133 & TAF1 & chrX & 71366239 & 71532374 &
amp\tabularnewline
257591 & ENSG00000085224 & ATRX & chrX & 77504878 & 77786233 &
amp\tabularnewline
259151 & ENSG00000147202 & DIAPH2 & chrX & 96684663 & 97604997 &
amp\tabularnewline
262571 & ENSG00000174460 & ZCCHC12 & chrX & 118823824 & 118826968 &
amp\tabularnewline
263741 & ENSG00000101972 & STAG2 & chrX & 123960212 & 124422664 &
amp\tabularnewline
264151 & ENSG00000102038 & SMARCA1 & chrX & 129446501 & 129523500 &
amp\tabularnewline
265031 & ENSG00000156531 & PHF6 & chrX & 134373253 & 134428791 &
amp\tabularnewline
268991 & ENSG00000196924 & FLNA & chrX & 154348524 & 154374638 &
amp\tabularnewline
\bottomrule
\end{longtable}

\begin{quote}
Bailey et al. (2018): Comprehensive Characterization of Cancer Driver
Genes and Mutations. Cell 173 (2), 371-385.e18. DOI:
10.1016/j.cell.2018.02.060.;
\end{quote}

\begin{quote}
Papaemmanuil et al. (2016): Genomic Classification and Prognosis in
Acute Myeloid Leukemia. The New England journal of medicine 374 (23), S.
2209--2221. \url{DOI:10.1056/NEJMoa1516192}.
\end{quote}

\hypertarget{karyotype-overview}{%
\subsection{Karyotype overview:}\label{karyotype-overview}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-10-1.pdf}

\hypertarget{chromosome-plots}{%
\subsection{Chromosome Plots:}\label{chromosome-plots}}

\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-1.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-2.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-3.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-4.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-5.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-6.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-7.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-8.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-9.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-10.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-11.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-12.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-13.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-14.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-15.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-16.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-17.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-18.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-19.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-20.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-21.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-22.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-23.pdf}
\includegraphics{Report_coriander_files/figure-latex/unnamed-chunk-11-24.pdf}

\hypertarget{the-legend-contains-two-characteristics}{%
\paragraph{The legend contains two
characteristics:}\label{the-legend-contains-two-characteristics}}

\begin{quote}
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Giemsa stain results. Recognized stain values: gneg, gpos50, gpos75,
  gpos25, gpos100, acen, gvar, stalk.
\item
  The abnormal regions (add or del).
\end{enumerate}
\end{quote}


\end{document}
